pyvrp/tests/test_Solution.py

```diff
@@ -819,3 +820,28 @@ def test_route_can_be_pickled():
         after_pickle = pickle.loads(bytes)
 
         assert_equal(after_pickle, before_pickle)
+
+
+@mark.parametrize(
+    ("assignment", "expected"), [((0, 0), 0), ((0, 1), 10), ((1, 1), 20)]
+)
+def test_fixed_vehicle_cost(assignment: Tuple[int, int], expected: int):
+    """
+    Tests that the solution tracks the total fixed vehicle costs of the
+    vehicles used for its routes.
+    """
+    data = read("data/OkSmall.txt")
+    data = make_heterogeneous(
```

Comment:
Probably yes, but at this point it has so much inertia I don't think we can get rid of it that easy. It'd help if we can at least standardise a data format for vehicle types, because then we can always revisit this later.

Reply:
I do think `make_heterogeneous` is a poor name for what it does. `replace_vehicle_types` seems better, but is quite verbose.

We might want to do what [`datetime`](https://docs.python.org/3/library/datetime.html) does: objects in that module expose a `replace()` method that can be called to replace aspects of the object. All `datetime` objects are immutable, so it returns a copy with the requested elements replaced by the arguments. Something like `ProblemData.replace` would fit in that overall setting.

Reply:
That's similar to what you already had implemented in #334, but then from the actual object. Should not be too hard to do.

I changed my position on this a bit because I spent a lot of time working with datetime last month, and it isn't that bad. But we should do this _on the object_, rather than as a separate helper.